### ISSUE TITLE
Cannot match object type pattern against non-object.

### ISSUE THREAD

**Issue by olwmc** [lutzhamel/asteroid#57]
Example:
```
load "io".

structure A with
	data x.
	end

println(A(0) is %A).    -- This works fine
println(3 is %integer). -- This also works fine
println(3 is %A).       -- This causes an error
```

**Comment by olwmc** [comment]
I'll work on a fix for this

**Comment by olwmc** [comment]
Fixed with #63

**Comment by olwmc** [comment]
Fixed with #63
